src/apns_connection.erl

```diff
@@ -136,7 +144,8 @@ handle_info(_Info, State) ->
 -spec terminate( Reason :: (normal | shutdown | {shutdown, term()} | term())
                , State  :: map()
                ) -> term().
-terminate(_Reason, _State) ->
+terminate(_Reason, #{gun_connection := GunConn}) ->
+  ok = gun:shutdown(GunConn),
```

Comment:
You don't need to check the result of this here. If it fails, there is nothing you can do at this point.